tests/s25Main/integration/testGamePlayer.cpp

```diff
@@ -62,8 +62,22 @@ BOOST_FIXTURE_TEST_CASE(ProductivityStats, WorldFixtureEmpty1P)
     BOOST_TEST(buildingRegister.CalcProductivities() == expectedProductivity, per_element());
     BOOST_TEST(buildingRegister.CalcAverageProductivity() == 0u);
 
+    std::vector<BuildingType> buildingTypes;
+    const auto buildingTypesEnum = helpers::EnumRange<BuildingType>{};
+
+    for(auto it = buildingTypesEnum.begin(); it != buildingTypesEnum.end(); ++it)
+    {
+        buildingTypes.push_back(*it);
+    }
+
+    // Sort buildings so military buildings are created first 
+    // and no buildings are destroyed when borders are recalculated
+    std::sort(buildingTypes.begin(), buildingTypes.end(), [](BuildingType a, BuildingType b) {
+        return !BuildingProperties::IsUsual(a) > !BuildingProperties::IsUsual(b);
+    });
```

Comment:
Neat but a bit hard to read. I guess `!BuildingProperties::IsUsual(a) && BuildingProperties::IsUsual(b)` would also work?

But even better: You want all "!IsUsual" buildings first, correct? I.e. partition the vector:
```suggestion
    std::partition(buildingTypes.begin(), buildingTypes.end(), [](BuildingType bld) { return !BuildingProperties::IsUsual(bld); });
```

Reply:
This works well